### PR TITLE
Refresh token

### PR DESCRIPTION
**Pull request by TEdwardJ**

### CODE REVIEW

File: src/main/java/com/vinylteam/vinyl/entity/TokenPair.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import lombok.Builder;
+import lombok.Value;
+
+@Value
+@Builder
+public class TokenPair {
+
+    String id;
+    String jwtToken;
+    String refreshToken;
```
Comment (littlepony99) [inline src/main/java/com/vinylteam/vinyl/entity/TokenPair.java:12]: add empty row after last class row

File: src/main/java/com/vinylteam/vinyl/security/LogoutTokenStorageService.java [hunk 2]
```diff
@@ -3,7 +3,8 @@
 import java.time.LocalDateTime;
 
 public interface LogoutTokenStorageService {
-    void storeToken(String token, LocalDateTime expirationDate);
 
-    boolean isTokenBlocked(String token);
+    boolean isTokenPairBlocked(String pairIdentifier);
+
+    void storePairIdentifier(String pairIdentifier, LocalDateTime expirationDate);
```
Comment (littlepony99) [inline src/main/java/com/vinylteam/vinyl/security/LogoutTokenStorageService.java:9]: add empty row after last class row

File: src/main/java/com/vinylteam/vinyl/security/SecurityConfiguration.java [hunk 3]
```diff
@@ -142,7 +142,7 @@ public LogoutService logoutService(JwtService jwtService,
     protected DelegatingFilterProxy getJwtValidatorFilterDelegatingProxy() {
         DelegatingFilterProxy delegateFilterProxy = new DelegatingFilterProxy();
         delegateFilterProxy.setServletContext(context);
-        delegateFilterProxy.setTargetBeanName("jwtValidationFilter");
+        delegateFilterProxy.setTargetBeanName("jwtValidatorFilter");
```
Comment (littlepony99) [inline src/main/java/com/vinylteam/vinyl/security/SecurityConfiguration.java]: move to a constant variable?

File: src/main/java/com/vinylteam/vinyl/service/DefaultProfileManagementService.java [hunk 4]
```diff
@@ -43,7 +45,7 @@ public ChangePasswordResponse changeProfilePassword(HttpServletRequest request,
         userService.changeUserPassword(userProfileInfo, user);
         String token = jwtService
                 .authenticateByRequest(new LoginRequest(user.getEmail(), userProfileInfo.getNewPassword()))
-                .getToken();
+                .getJwtToken();
         logoutService.logout(request, null, null);
```
Comment (littlepony99) [inline src/main/java/com/vinylteam/vinyl/service/DefaultProfileManagementService.java:49]: seems to me that its not a good "interface" idea
maybe you should create logout method with 1 param, which under hood will send request logoutService.logout(request, null, null);

File: src/main/java/com/vinylteam/vinyl/service/JwtService.java [hunk 5]
```diff
@@ -1,28 +1,41 @@
...
+    String getPairIdentifier(String token);
+
     UserSecurityResponse getCheckResponseIfTokenValid(String token);
 
     UserSecurityResponse authenticateByRequest(LoginRequest loginRequest);
 
+    TokenPair getTokenPair(JwtUser user);
+
+    UserSecurityResponse refreshByToken(String refreshToken);
 }
```
Comment (littlepony99) [inline src/main/java/com/vinylteam/vinyl/service/JwtService.java:42]: add empty row after last class row

File: src/main/java/com/vinylteam/vinyl/service/impl/DefaultUserService.java [hunk 6]
```diff
@@ -26,9 +27,10 @@
 public class DefaultUserService implements UserService {
 
     private final UserDao userDao;
-    private final SecurityService securityService;
-    private final EmailConfirmationService emailConfirmationService;
 
+    private SecurityService securityService;
+
+    private final EmailConfirmationService emailConfirmationService;
```
Comment (littlepony99) [inline src/main/java/com/vinylteam/vinyl/service/impl/DefaultUserService.java:33]: add empty row, divide variable and method parts

**Review by littlepony99 — CHANGES_REQUESTED** [review]

File: src/main/java/com/vinylteam/vinyl/entity/Currency.java [hunk 7]
```diff
@@ -5,6 +5,8 @@
 
 import java.util.Optional;
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/entity/Currency.java:3]: please remove all logging statements from this enum

Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/entity/Currency.java:3]: could you please also check this code?

else if (" EUR".equals(currencyDescription) || "EUR".equals(currencyDescription) || "€".equals(currencyDescription)
                || "&nbsp;€".equals(currencyDescription)) {
            resultingCurrency = EUR;
        }

is white space between " and EUR ok ?

" EUR .."

File: src/main/java/com/vinylteam/vinyl/entity/TokenPair.java [hunk 8]
```diff
@@ -0,0 +1,13 @@
+package com.vinylteam.vinyl.entity;
+
+import lombok.Builder;
+import lombok.Value;
+
+@Value
+@Builder
+public class TokenPair {
+
+    String id;
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/entity/TokenPair.java:10]: why package-private modifiers?

File: src/main/java/com/vinylteam/vinyl/entity/UniqueVinyl.java [hunk 9]
```diff
@@ -27,16 +28,11 @@
     private String imageLink;
 
     @Getter(AccessLevel.NONE)
-    @Setter(AccessLevel.NONE)
     @Field(type = FieldType.Boolean)
     @EqualsAndHashCode.Exclude
-    private boolean offers;
+    private boolean hasOffers;
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/entity/UniqueVinyl.java:33]: please revert this
its should be "offers"
we use has* for boolean's getters in this case

File: src/main/java/com/vinylteam/vinyl/security/InMemoryLogoutTokenService.java [hunk 10]
```diff
@@ -27,7 +27,7 @@ public void initCache() {
     }
 
     private Expiry<String, LocalDateTime> getExpirationStrategy() {
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/security/InMemoryLogoutTokenService.java:39]: please move this private method below public ones

Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/security/InMemoryLogoutTokenService.java:39]: could you please also fix all parameters String s in this class?
you could go to the documentation of com.github.benmanes.caffeine.cache.Expiry and read what s means

**Review by hemiSRT8 — CHANGES_REQUESTED** [review]

File: src/main/java/com/vinylteam/vinyl/entity/Currency.java (hunk 7 above)
Comment (TEdwardJ) [reply src/main/java/com/vinylteam/vinyl/entity/Currency.java:3]: Yes, I suppose the author wanted to be sure that any possible string with currency will be recognized correctly, but think trim() will be better

File: src/main/java/com/vinylteam/vinyl/entity/TokenPair.java (hunk 8 above)
Comment (TEdwardJ) [reply src/main/java/com/vinylteam/vinyl/entity/TokenPair.java:10]: @FieldDefaults(makeFinal=true, level=AccessLevel.PRIVATE)  - defined in the @Value annotation

Comment (TEdwardJ) [reply src/main/java/com/vinylteam/vinyl/entity/TokenPair.java:10]: @FieldDefaults(makeFinal=true, level=AccessLevel.PRIVATE) from @Value annotation from lombok

File: src/main/java/com/vinylteam/vinyl/entity/UniqueVinyl.java (hunk 9 above)
Comment (TEdwardJ) [reply src/main/java/com/vinylteam/vinyl/entity/UniqueVinyl.java:33]: it requires reloading data to ElasticSearch... or probably there is another solution, but it still has to be found. Just renaming makes the catalog page fails

**Review by littlepony99 — APPROVED** [review]

**Review by hemiSRT8 — APPROVED** [review]

**Pull request merged**